### DISCUSSION TITLE
cisco ios logs not working

### DISCUSSION THREAD

**Discussion by Jackson-Pollock** [Security-Onion-Solutions/securityonion#7854] [Unsupported Versions] [answered]
I've `.log` files of cisco ios switch/router and I wanted to ingest those logs into ES before testing a real syslog server.  I've setup filebeat in one of the Ubuntu machine and enabled cisco ios filebeat module and then copied those `.log` files into the `var.paths` and as I can see in the log file, filebeat is sending those logs to the Logstash of Security Onion but I unable to see those logs into Kibana.

**Comment by Jackson-Pollock** [comment, marked as answer]
As I found this in logstash logs: snipped
`[WARN ][logstash.outputs.elasticsearch] Could not index event to Elasticsearch. "reason"=>"pipeline with id [filebeat-8.1.3-cisco-ios-pipeline] does not exist"}}}}`
Is it because I used filebeat from ES officials and not from SO Downloads section? But that do not have cisco module built in.

**Reply by Jackson-Pollock** [reply]
This is solved by installing Filebeat 7.17.1

**Comment by TotieBash** [comment]
hmm, I'm trying to accomplish the same thing enabling Filebeat cisco-ios module to send my Cisco IOS logs. However I am trying to do it using SO's filebeat docker container.  

It should look like this.
CISCO-IOS -> SO:9002/udp

As I understand your config, it looks like this.  Correct?
CISCO-IOS -> Filebeat:9002/udp -> SO:514/udp

I was inspired by this Netflow video my Matt Gracie
https://youtu.be/ew5gtVjAs7g

However, my config is still not working.  I followed the video instruction to the T but no joy.  I will lab your configuration just for testing purpose.  


My /opt/so/saltstack/local/pillar/minions/SO_standalone.sls looks like this:
filebeat:
  third_party_filebeat:
    modules:
      cisco:
        ios:
          enabled: true
          var.syslog_host: 0.0.0.0
          var.syslog_port: 9002
      netflow:
        log:
          enabled: true
          var.netflow_host: 0.0.0.0
          var.netflow_port: 2055

**Reply by Jackson-Pollock** [reply]
I saw the video but my testing was different than this. I used filebeat with cisco ios module on another machine to forward .log file to the so-logstash. There is default pipeline available for this hence it worked easily. In next step I am going to let SO receive logs over default syslog and then use so-filebeat modules by taking reference from this vdo. Thanks for sharing!

**Reply by dougburks** [reply]
@TotieBash If you have questions or problems, please start your own discussion rather than replying to somebody else's discussion.
https://github.com/Security-Onion-Solutions/securityonion/discussions/1720
